src/app/accounts/update-account-custom-fields.ts

```diff
@@ -0,0 +1,48 @@
+import { getAccountsConfig } from "@config"
+
+import { CouldNotFindError } from "@domain/errors"
+import { NoAccountCustomFieldsError } from "@domain/accounts"
+
+import { AccountCustomFieldsRepository, AccountsRepository } from "@services/mongoose"
+
+const { customFields: customFieldsSchema } = getAccountsConfig()
+
+const defaultValues = customFieldsSchema.reduce((acc, val) => {
+  if (val.defaultValue !== undefined) {
+    acc[val.name] = val.defaultValue
+  }
+  return acc
+}, {} as { [k: string]: AccountCustomFieldValues })
+
+export const updateAccountCustomFields = async ({
+  accountId,
+  modifiedByUserId,
+  customFields,
+}: {
+  accountId: AccountId
+  modifiedByUserId: UserId
+  customFields: { [k: string]: AccountCustomFieldValues }
+}): Promise<AccountCustomFields | ApplicationError> => {
+  if (!customFieldsSchema || customFieldsSchema.length <= 0)
+    return new NoAccountCustomFieldsError()
+
+  const accountCustomFieldsRepo = AccountCustomFieldsRepository()
+
+  const account = await AccountsRepository().findById(accountId)
+  if (account instanceof Error) return account
+
+  const accountCustomFields = await accountCustomFieldsRepo.findById(account.id)
+  const isEmpty = accountCustomFields instanceof CouldNotFindError
+  if (accountCustomFields instanceof Error && !isEmpty) return accountCustomFields
+
+  let data = customFields
+  if (!isEmpty) {
+    data = { ...accountCustomFields.customFields, ...customFields }
+  }
+
+  return accountCustomFieldsRepo.persistNew({
+    accountId: account.id,
+    modifiedByUserId,
+    customFields: { ...defaultValues, ...data },
```

Comment:
we're using defaultFields in two places. here (on write), but also https://github.com/GaloyMoney/galoy/pull/1465/files#diff-f5f42167109591eaca82e92d001bdd96f4db522ee0da89816e3ec9a56e7ff15bR25, on read. 

I'm wondering if this is a good behavior. 

I've never really design any database system like this, but my intuition tells me that default value would either be on read, or write, but not both.

Reply:
one way to go with it is to not have any default settings. maybe that would be an option

at least for the current use case, I don't see a need to have default values

Reply:
IMO defaults should be set on write (not read). But agree to keep the implementation to the minimum for the use case at hand.

Reply:
we need default values and also we need them on read because possible updates on required fields. Please check Sam's comment  https://github.com/GaloyMoney/galoy/pull/1465#discussion_r932725226

Reply:
I'm not sure I agree with Sam input, my response here: https://github.com/GaloyMoney/galoy/pull/1465#discussion_r949023827